File: chrome/test/data/webui/glic/browser_tests/glic_focus_interactive_uitest.ts
// Copyright 2026 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import {ApiTestFixtureBase, runUntil, testMain} from './browser_test_base.js';

class GlicFocusInteractiveTest extends ApiTestFixtureBase {
  override async setUpTest() {
    await this.client.waitForFirstOpen();
  }

  async testFocusOnSidePanelOpen() {
    await runUntil(() => document.activeElement?.id === 'inputBox');
    await runUntil(() => document.hasFocus());
  }
}

const TEST_FIXTURES = [
  GlicFocusInteractiveTest,
];

testMain(TEST_FIXTURES);

File: chrome/browser/glic/host/glic_focus_interactive_uitest.cc
// Copyright 2026 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "base/test/bind.h"
#include "chrome/browser/glic/test_support/glic_api_test.h"
#include "chrome/browser/ui/browser.h"
#include "chrome/browser/ui/browser_element_identifiers.h"
#include "chrome/browser/ui/interaction/browser_elements.h"
#include "chrome/browser/ui/views/interaction/browser_elements_views.h"
#include "chrome/test/interaction/interactive_browser_test.h"
#include "content/public/test/browser_test.h"

namespace glic {
namespace {

class GlicFocusInteractiveTest : public InteractiveGlicApiTest {
 public:
  GlicFocusInteractiveTest()
      : InteractiveGlicApiTest("./glic_focus_interactive_uitest.js") {}
};

// Regression test for b/475260887. The autofocus <input> element in the client
// page does not receive focus on opening the side panel.
IN_PROC_BROWSER_TEST_F(GlicFocusInteractiveTest, testFocusOnSidePanelOpen) {
  RunTestSequence(OpenGlic());
  ExecuteJsTest();
}

// Regression test for b/504144250. The client page in the side panel does not
// lose page focus.
IN_PROC_BROWSER_TEST_F(GlicFocusInteractiveTest, testBlurOnOmniboxFocus) {
  RunTestSequence(OpenGlic());
  ExecuteJsTest();
  RunTestSequence(Do(base::BindLambdaForTesting([&]() {
    views::View* omnibox =
        BrowserElementsViews::From(browser())->GetView(kOmniboxElementId);
    ASSERT_TRUE(omnibox);
    omnibox->RequestFocus();
  })));

  ContinueJsTest();
}

}  // namespace
}  // namespace glic
